src/Sequence.php

```diff
@@ -99,6 +99,27 @@ public function apply(callable $callback)
     }
 
     /**
+     * @param mixed $value
+     * @param null|int $num
+     */
+    public function fill($value, int $num = null)
+    {
+        if (null === $num) {
+            \array_walk($this->array, function (&$v) use ($value) {
+                $v = $value;
+            });
```

Comment:
Isn't this just: `$this->array = array_fill(0, count($this), $value);`? I don't think we need `array_walk` and reference here.